### DISCUSSION TITLE
Is it possible to return a ResponseEntity<T> from a resolver

### DISCUSSION THREAD

**Discussion by naveen4801** [graphql-java-kickstart/graphql-java-tools#589] [Q&A]
I'm using graphql-java with spring boot.
My schema looks like this.
type Query{
    userById(id: String): User
}
type User {
    id: String
    name: String
    email: String
}
I have a QueryResolver class that implements GraphQLQueryResolver. My get method inside this class looks like this.
public User getUserById(String id) {
   return someClient.getUserById(id).getBody(); //getUserById inside someClient returns a ResponseEntity<User>
}
What I need inside my QueryResolver is to return a ResponseEntity<User> instead of a User. So it'll look like the code below
public ResponseEntity<User>  getUserById(String id) {
   return someClient.getUserById(id); 
}
The above code returns a schema error saying org.springframework.http.ResponseEntity<User> cannot be mapped to a GraphQL type! Since GraphQL-Java deals with erased types at runtime, only non-parameterized classes can represent a GraphQL type. What does this mean? Is it possible to return a ResponseEntity<> from a resolver?

**Comment by vojtapol** [comment]
>  Is it possible to return a `ResponseEntity<>` from a resolver?

No that is not possible. Could you elaborate why returning `User` directly does not work for your use case?

The only other type that is supported as a return value is `DataFetcherResult`.

**Comment by naveen4801** [comment]
Currently We are migrating REST services into GraphQL , Since we were retrieving errors and responses in ResponseEntity<T> type from AOP, so it will be more flexible to get response from GraghQL in ResponposeEntity<T> type

**Comment by naveen4801** [comment]
can you please elaborate how can i change this  into a DataFetcherResult from ResponseEntity<T>

**Comment by rever67697** [comment]
```java

public  DataFetcherResult<SomeData> getUserById(){
    return DataFetcherResult.newResult().data(data).error(error).build();
}

```
